eng/common/scripts/stress-testing/find-all-stress-packages.ps1

```diff
@@ -62,6 +69,31 @@ function GetUsername() {
     return $stressUser.ToLower()
 }
 
+function ScenariosMatrixGeneration(
+    [string]$matrixFilePath
+) {
+    $yamlConfig = Get-Content $matrixFilePath -Raw
+    $matrix = GenerateMatrix (GetMatrixConfigFromYaml $yamlConfig) "sparse"
+    $serializedMatrix = SerializePipelineMatrix $matrix
+    $prettyMatrix = $serializedMatrix.pretty | ConvertFrom-Json -AsHashtable
+
+    $scenariosMatrix = @()
+    foreach($permutation in $prettyMatrix.GetEnumerator()) {
+        $entry = @{}
+        $entry.Name = $permutation.key
+        foreach ($param in $permutation.value.GetEnumerator()) {
+            $entry.add($param.key, $param.value)
+        }
+        $scenariosMatrix += $entry
```

Comment:
Is there a reason re-using the permutation object wouldn't work?

```suggestion
        $permutation.Name = $permutation.key
        $scenariosMatrix += $permutation
```

Reply:
Permutation is of type System.Collections.DictionaryEntry which doesn't have some of the methods that I needed